src/Blocks/BlockFarmland.h

```diff
@@ -101,9 +128,11 @@ class cBlockFarmlandHandler final :
 		}
 
 		// Check whether we should revert to dirt:
+		// TODO: fix for signs and slabs (possibly more blocks) - they should destroy farmland
 		auto upperBlock = a_ChunkInterface.GetBlock(a_BlockPos.addedY(1));
 		if (cBlockInfo::FullyOccupiesVoxel(upperBlock))
 		{
+			// TODO: somehow use TurnToDirt
```

Comment:
DoWithChunkAt on the interface?

Reply:
Done

Reply:
Note: right now, it's only for testing that it works correctly. It looks like the current implementation of `OnNeighborChanged` won't destroy farmland after placing any non-solid block (signs, anything else?), so there's no risk of any entity getting stuck or falling though farmland anyway